src/client/src/ui/blotter/components/BlotterHeader.tsx

```diff
@@ -48,66 +44,57 @@ const Fill = styled.div`
   height: 1rem;
 `
 
-export default class BlotterHeader extends Component<Props, State> {
-  state = {
-    quickFilterText: '',
-  }
-
-  render() {
-    const { canPopout, onPopoutClick, onExportToExcelClick } = this.props
-    const { quickFilterText } = this.state
-
-    return (
-      <BlotterHeaderStyle>
-        <BlotterLeft>Executed Trades</BlotterLeft>
-        <BlotterRight>
-          <ExcelButton onClick={onExportToExcelClick} />
-          <BlotterToolbar
-            isQuickFilterApplied={quickFilterText.length !== 0}
-            quickFilterChangeHandler={this.quickFilterChangeHandler}
-            removeQuickFilter={this.removeQuickFilter}
-            removeAllFilters={this.removeAllFilters}
-            removeFilter={this.removeFilter}
-            filterModel={this.props.gridApi ? this.props.gridApi.getFilterModel() : null}
-            columnDefinitions={columnDefinitions}
-          />
-          {canPopout && (
-            <React.Fragment>
-              <Fill />
-              <BlotterControls onClick={onPopoutClick}>
-                <PopoutIcon width={0.8125} height={0.75} />
-              </BlotterControls>
-            </React.Fragment>
-          )}
-        </BlotterRight>
-      </BlotterHeaderStyle>
-    )
-  }
+const BlotterHeader: FC<Props> = ({ gridApi, canPopout, onExportToExcelClick, onPopoutClick }) => {
+  const [quickFilterText, setQuickFilterText] = useState('')
 
-  private quickFilterChangeHandler = (event: React.FormEvent<HTMLInputElement>) => {
-    const { gridApi } = this.props
-    const { value } = event.currentTarget
-    this.setState({ quickFilterText: value })
-    return gridApi && gridApi.setQuickFilter(value)
-  }
+  const quickFilterChangeHandler = useCallback(
+    event => {
+      setQuickFilterText(event.currentTarget.value)
+      return gridApi && gridApi.setQuickFilter(event.currentTarget.value)
+    },
+    [setQuickFilterText, gridApi],
```

Comment:
I still wonder if `gridApi && gridApi.setQuickFilter` should be included in the deps, but that would only occur if `gridApi` was the same object, but the `setQuickFilter` changed, which would be a horrible API to work with.

Leave it as-is - we can investigate later if this was to ever cause an issue.

Reply:
React guarantees `setState`  function's identity and it won't change on re-renders, so it can be omitted from the dependency list. 
However `gridApi` should stay .